src/types/singleModel.js

```diff
@@ -0,0 +1,35 @@
+/* @flow */
+
+'use strict';
+
+const BaseType = require('./base');
+
+class SingleModel extends BaseType {
+
+    constructor (parentModel: Object, Model: Function, options: Object = {}) {
+        super(parentModel, options);
+
+        Object.assign(this, {
+            _ModelConstructor: Model,
+            _model: new Model(parentModel),
```

Comment:
I'll need to dig into this because I'm using the model to emit events up the chain, which we need to keep in HasOne